pyravendb/raven_operations/timeseries_operations.py

```diff
@@ -0,0 +1,161 @@
+from typing import Optional, Iterable, List, Generator
+from pyravendb.commands.raven_commands import *
+from pyravendb.tools.utils import Utils
+from .operations import Operation
+from datetime import datetime
+
+import sys
+
+
+class TimeSeriesRange:
+    def __init__(self, name, from_date: Optional[datetime] = None, to_date: Optional[datetime] = None):
+        if not name:
+            raise ValueError("TimeSeriesRange must include name")
+        self.name = name
+        self.from_date = from_date
+        self.to_date = to_date
+
+
+class GetTimeSeriesOperation(Operation):
+    def __init__(self, document_id: str, ranges: Iterable[TimeSeriesRange] or TimeSeriesRange = None,
+                 start: int = 0, page_size: int = sys.maxsize):
+        """
+        Build the time series get operations construct with ranges or from_date and to_date.
+        """
+        super().__init__()
+        if isinstance(ranges, TimeSeriesRange):
+            ranges = [ranges]
+
+        self._ranges = ranges
+        self._document_id = document_id
+        self._start = start
+        self._page_size = page_size
+
+    def get_command(self, store, conventions, cache=None):
+        return self._GetTimeSeriesCommand(self._document_id, self._ranges, self._start, self._page_size)
+
+    class _GetTimeSeriesCommand(RavenCommand):
+        def __init__(self, document_id: str, ranges: Iterable[TimeSeriesRange],
+                     start: int = 0, page_size: int = sys.maxsize):
+            super().__init__(method="GET", is_read_request=True)
+
+            if not document_id:
+                raise ValueError("Invalid document_id")
+
+            self._document_id = document_id
+            self._ranges = ranges
+            self._start = start
+            self._page_size = page_size
+
+        def create_request(self, server_node):
+            self.url = (f"{server_node.url}/databases/{server_node.database}"
+                        f"/timeseries?id={Utils.quote_key(self._document_id)}"
+                        f"{f'&start={self._start}' if self._start > 0 else ''}"
+                        f"{f'&pageSize={self._page_size}' if self._page_size < sys.maxsize else ''}")
+
+            if self._ranges is not None:
+                for range_ in self._ranges:
+                    self.url += (f"&name={Utils.quote_key(range_.name)}"
+                                 f"&from={Utils.datetime_to_string(range_.from_date)}"
+                                 f"&to={Utils.datetime_to_string(range_.to_date)}")
```

Comment:
If the user passed `None`, what do we get here?

Reply:
leftovers fixed, ranges are mandatory.

Reply:
you can give only a name and it still works.
the server will return with all the time-series he has with the name